docs/development/autoscaling-specifics-for-components.md

```diff
@@ -0,0 +1,69 @@
+
+- `HVPA`
+
+   In `HVPA` mode, the etcd is scaled by the [hvpa-controller](https://github.com/gardener/hvpa-controller). The gardenlet/gardener-operator is creating an `HVPA` resource for the etcd (`main` or `events`).
+   The `HVPA` enables a vertical scaling for etcd.
+
+   The `HVPA` mode is the used autoscaling mode when the `HVPA` feature gate is enabled (and the `VPAForETCD` feature gate is disabled).
+
+- `VPA`
+
+   In `VPA` mode, the etcd is scaled by a native `VPA` resource.
+
+   The `VPA` mode is the used autoscaling mode when the `VPAForETCD` feature gate is enabled (takes precedence over the `HVPA` feature gate).
+
+For both of the autoscaling modes downscaling is handled more pessimistically to prevent many subsequent etcd restarts. Thus, for `production` and `infrastructure` Shoot clusters (or all Garden clusters), downscaling is deactivated for the main etcd. For all other Shoot clusters, lower advertised requests/limits are only applied during the Shoot's maintenance time window.
+
+## Shoot Kubernetes API Server Autoscaling
+
+There are three supported autoscaling modes for the Shoot Kubernetes API Server.
+
+- `Baseline`
+
+   In `Baseline` mode, the Shoot Kubernetes API server is scaled by active HPA and VPA in passive, recommend-only mode.
+
+   The API server resource requests are computed based on the Shoot's minimum Nodes count:
+   | Range       | Resource Requests |
+   |-------------|-------------------|
+   | [0, 2]      | `800m`, `800Mi`   |
+   | (2, 10]     | `1000m`, `1100Mi` |
+   | (10, 50]    | `1200m`, `1600Mi` |
+   | (50, 100]   | `2500m`, `5200Mi` |
+   | (100, inf.) | `3000m`, `5200Mi` |
+
+   The `Baseline` mode is the used autoscaling mode when the `HVPA` and `VPAAndHPAForAPIServer` feature gates are not enabled.
+
+- `HVPA`
+
+   In `HVPA` mode, the Shoot Kubernetes API server is scaled by the [hvpa-controller](https://github.com/gardener/hvpa-controller). The gardenlet is creating an `HVPA` resource for the API server. The `HVPA` resource is backed by HPA and VPA both in recommend-only mode. The hvpa-controller is responsible for enabling simultaneous horizontal and vertical scaling by incorporating the recommendations from the HPA and VPA.
+
+   The initial API server resource requests are `500m` and `1Gi`.
+
+   The `HVPA` mode is the used autoscaling mode when the `HVPA` feature gate is enabled (and the `VPAAndHPAForAPIServer` feature gate is disabled).
+
+- `VPAAndHPA`
+
+   In `VPAAndHPA` mode, the Shoot Kubernetes API server is scaled simultaneously by VPA on CPU and memory utilization and by HPA - on CPU and memory usage. The gardenlet configures VPA and HPA resources in a such a way that the VPA's `maxAllowed` CPU and memory values are a little smaller than the HPA's average usage target. This allows VPA to scale vertically on the Pod's CPU and memory requests. Once all Pods on average exceed the maxAllowed CPU/memory, HPA is scaling horizontally (by adding a new replica).
```

Comment:
This doesn't seem correct:
VPA's `maxAllowed` is _higher_ than the HPA's average usage target. VPA has `CPU: 7`, `Memory: 28G`, HPA has `CPU: 6`, `Memory: 24G`. Which makes sense, given that on a fully utilized Node, we would still want to get a signal that horizontal scaling should be done.

Reply:
Yes, HPAs averateUtilization target is 10% lower than VPA's maxAllowed values (and not vice-versa as written above) 🙈 .

I changed the section to:
>  In `VPAAndHPA` mode, the Shoot Kubernetes API server is scaled simultaneously by VPA and HPA on the same metric (CPU and memory usage). The pod-trashing cycle between VPA and HPA scaling on the same metric is broken by configuring the HPA to scale on average usage (not on average utilization) and by picking the target average utilization value in sync with VPA's allowed maximums. This makes possible VPA to first scale vertically on CPU/memory usage. Once all Pods' average CPU/memory usage is close to exceed the VPA's allowed maximum CPU/memory (the HPA's target average utilization, 10-20% less than VPA's allowed maximums), HPA is scaling horizontally (by adding a new replica).